solc/CommandLineParser.cpp

```diff
@@ -927,13 +927,22 @@ General Information)").c_str(),
 	else
 		m_options.input.mode = InputMode::Compiler;
 
-	if (
-		m_args.count(g_strExperimentalViaIR) > 0 &&
-		m_options.input.mode != InputMode::Compiler &&
-		m_options.input.mode != InputMode::CompilerWithASTImport
-	)
+	map<string, vector<InputMode>> validOption_InputModeCombinations = {
```

Comment:
I'd use a `set` here. Checking if option is in there will be more efficient and also more concise to write (you can just call `count()` on the set).
```suggestion
	map<string, set<InputMode>> validOption_InputModeCombinations = {
```